[Term entry] Python pillow .putpixel() method (#4731)
Codecademy/docs: content/pillow/concepts/image/terms/putpixel/putpixel.md

> ---
> Title: '.putpixel()' 
> Description: 'The putpixel() method in Python's Pillow library (PIL) is used to put a pixel in a specific place on an image. It takes two parameters, one being the position (x, y) and the other being the color.'
> Subjects: 
>   - 'Computer science'
>   - 'Data Visualization'
>   - 'Game development'
> Tags: 
>   - 'Images'
>   - 'Value'
>   - 'Computer vision'
>   - 'Pillow'
> CatalogContent: 
>   - 'learn-python-3'
>   - 'paths/computer-science'
> ---
> ## .putpixel() Method

Reviewer: ```suggestion
---
Title: '.putpixel()' 
Description: 'Sets the color of a specific pixel in an image to a given value. '
Subjects: 
  - 'Computer Science'
  - 'Data Science'
  - 'Data Visualization'
Tags: 
  - 'Computer Vision'
  - 'Images'
  - 'Pillow'
  - 'Values'
CatalogContent: 
  - 'learn-python-3'
  - 'paths/computer-science'
  - 'paths/data-science'
  - 'paths/data-science-foundations'
---

```

Reviewer: The subjects ad=nd tags should be in alphabetical order and the description should be what the term does.

The **.putpixel()** method is used to insert pixels onto an image. While it is mostly used to set the color of an individual pixel, it can also be used for other purposes such as changing pixel transparency or creating different shapes within the image.

> 

Reviewer: reframed the sentence

## Syntax

```pseudo
Image.putpixel(xy, color)
```

> **xy:** Specifies the coordinates of the pixel where the color will be placed, x representing the horizontal position, and y representing the vertical position.
> 
> **color:** Specify the color that you want the pixel to be set as. This can be in various forms inside a tuple such as RGB or an integar that specifies the color index.

Reviewer: ```suggestion
- `xy`: Specifies the coordinates of the pixel where the color will be placed, x representing the horizontal position, and y representing the vertical position.
- `color`: Specifies the color for the pixel, which can be:
  - RGB tuple (3 integers) for true color.
  - RGBA tuple (4 integers) for true color with transparency.
  - Integer for color index in indexed color images ("P" mode).
  - Single integer or tuple with a single integer for grayscale images.
  
```

Reviewer: Added the ways colors can be specified

## Example

```py
from PIL import Image

width, height = 200, 200
img = Image.new('RGB', (width, height), color='black')

#Adds the blue pixel on image
blue_pixel_position = (50, 50)
blue_color = (0, 0, 255)

#Adds the red pixel on image
red_pixel_position = (100, 100)
red_color = (255, 0, 0)

#Adds the green pixel on image
green_pixel_position = (150, 150)
green_color = (0, 255, 0)

img.putpixel(red_pixel_position, red_color)
img.putpixel(green_pixel_position, green_color)
img.putpixel(blue_pixel_position, blue_color)

> img.save("output_image.png")

Reviewer: ```suggestion
img.show()
```

Reviewer: Let us show it instead of saving it.

```

```shell
python putpixel.py

> ```

Reviewer: For this block, use the line "The above code generates the output image as follows:"

and add the output image in the "media" folder in docs, and link it here

Author: ![output_image](https://github.com/Codecademy/docs/assets/169033590/1c41d738-e226-4673-9c9d-fe77a568779e)

---
Title: 'putpixel' 
Description: 'The putpixel() method in Python's Pillow library (PIL) is used to put a pixel in a specific place on an image. It takes two parameters, one being the position (x, y) and the other being the color.' 
Subjects: 

>   - 'Web development'
>   - 'Data Visualization'
>   - 'Game development'

Reviewer 2: ```suggestion
  - 'Computer Science'
  - 'Data Visualization'
  - 'Game development'
```

Tags: 
  - 'Images'
  - 'Value'
  - 'tuple'
CatalogContent: 
  - 'learn-python-3'
  - 'paths/web-development'
---

# .putpixel() Method

The **putpixel** method is used to insert pixels onto an image. While its mostly used to set the color of an individual pixel, it can also be used for other purposes such as changing pixel transparency or creating different shapes within the image.

## Syntax

```python
Image.putpixel(xy, color)
```
**xy:** Specifies the coordinates of the pixel where the color will be placed, x representing the horizontal position, and y representing the vertical position.

**The 'color' parameter specifies the color that you want the pixel to be set as.** This can be in various forms inside a tuple such as RGB or an integar which specifies the color index.

> ## Example
> ```python

Reviewer 2: ```suggestion
## Example

```py
```

from PIL import Image

width, height = 200, 200
img = Image.new('RGB', (width, height), color='black')

#Adds the blue pixel on image
blue_pixel_position = (50, 50)
blue_color = (0, 0, 255)

#Adds the red pixel on image
red_pixel_position = (100, 100)
red_color = (255, 0, 0)

#Adds the green pixel on image
green_pixel_position = (150, 150)
green_color = (0, 255, 0)

img.putpixel(red_pixel_position, red_color)
img.putpixel(green_pixel_position, green_color)
img.putpixel(blue_pixel_position, blue_color)

img.save("output_image.png")
```

```shell command
python putpixel.py

> ```

Reviewer 2: Also, add the image showing the output

Author: ![output_image](https://github.com/Codecademy/docs/assets/169033590/0ae96cdc-ad69-458b-8016-08ec368cd51a)
You will have to zoom in a fair bit to see the output but there are 3 coloured dots on there.

Author: ![output_image](https://github.com/Codecademy/docs/assets/169033590/4e251b77-fcf3-4a5d-ade1-bb35b442cbf0)
